src/rtctools/optimization/optimization_problem.py

```diff
@@ -314,6 +314,24 @@ def solver_success(
         if log_level == logging.ERROR and not log_solver_failure_as_error:
             log_level = logging.INFO
 
+        if self.solver_options()["solver"].lower() in ["knitro"]:
+            list_feas_flags = [
```

Comment:
Checking if an element exists is faster ( O(1) instead of O(N) ) if you make this a set `{...}`